url4192078a2f_unsegnor_hipotetizador_tgz-pJ8-hipotetizador_MedirTiemposJ8/src/hipotetizador/Hipo.java

```diff
@@ -0,0 +1,1790 @@
+     * @param longitud
+     * @return
+     */
+    private ArrayList<InfoElemento> elaborar_tabla(Par<Long, Long>[][] _cuentas, float umbral, boolean[] entradas, int longitud) {
+        //Leer la historia y elaborar la tabla de frecuencias
+
+        //Aquí filtramos por umbral (que de momento no ha resultado ser útil)
+        //y por entradas
+
+        ArrayList<InfoElemento> tabla = new ArrayList<>();
+        for (int i = 0; i < _cuentas.length; i++) { //Para cada entrada
+            if (entradas[i]) { //Si es una entrada que queremos miramos sus elementos
+                for (int j = 0; j < _cuentas[i].length; j++) {
+                    Par<Long, Long> par = _cuentas[i][j];
+                    long apariciones = par.getPrimero();
+                    long total = par.getSegundo();
+                    long ausencias = total - apariciones;
+                    Elemento verdadero = new Elemento();
+                    //Si la frecuencia es mayor lo añadimos al conjunto
+                    if ((float) apariciones / (float) total > umbral) {
+                        verdadero.setEntrada(i);
+                        verdadero.setSubindice(j);
+                        verdadero.setVerdadero(true);
+                        InfoElemento infov = new InfoElemento();
+                        infov.setElemento(verdadero);
+                        infov.setApariciones(apariciones);
+                        infov.setTotal(total);
+                        tabla.add(infov);
+                    }
+
+                    if ((float) ausencias / (float) total > umbral) {
+                        Elemento falso = new Elemento();
+                        falso.setEntrada(i);
+                        falso.setSubindice(j);
+                        falso.setVerdadero(false);
+                        InfoElemento infof = new InfoElemento();
+                        infof.setElemento(falso);
+                        infof.setApariciones(ausencias);
+                        infof.setTotal(total);
+                        tabla.add(infof);
+                    }
+                }
+            }
+        }
+        return tabla;
+    }
+
+    /**
+     * Buscará una explicación a la totalidad de las acciones que sucedan en la
+     * historia, inventará todo el contexto oculto que sea necesario devolverá
+     * la explicación más sencilla
+     *
+     * @param umbral_de_hipotesis
+     * @param umbral_de_certeza
+     */
+    //TODO implementar algoritmo greedy para seleccionar los bits que nos inventaremos
+    //TODO ir probando con los diferentes casos y ver el aumento de explicabilidad que conseguimos
+    //guardar en una tabla las características de la regla y el aumento de explicabilidad conseguido a ver si vemos la regla
+    public Teoria sinAmbiguedad(int nentradas, int long_ventana, boolean[][] la_historia, double umbral_de_hipotesis, double umbral_de_certeza, float umbral_explicabilidad, boolean generar_subreglas) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 78 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 383-392 into a separate method.